### DISCUSSION TITLE
How do I grab the current page's parent name?

### DISCUSSION THREAD

**Discussion by LizThompson93** [squidfunk/mkdocs-material#7101] [Q&A] [answered]
Before I begin, I just want to thank you Martin (Squidfunk) and the team for your work on Material, its fantastic documentation, continued support, and how fast you respond. It's an amazing tool! Thank you.

For navigation and spacing purposes, I shorthanded the display names. It makes sense in the table of content but not quite so on navigation footer. I would like a way to reference the correct respective parent name and append it to the navigation footer text.
![image](https://github.com/squidfunk/mkdocs-material/assets/96261305/19dede0b-d444-49c5-8106-9858df28cc7e)
(-this is a public page I quickly found online for demonstration purposes)

I know I will have to override footer.html but I'm not sure what the selector/variable name is to fetch a parent name or a strategy to get it.

Thank you.

**Comment by squidfunk** [comment, marked as answer]
Thanks for the praise! I'm glad our work is useful to you ❤️

`page.ancestors` contains the list of ancestors in increasing levels, so `page.ancestors[0]` should be what you're looking for. Note that you should safe guard for the case when there are no ancestors, but something like the following should work, which means you should be able to integrate this into your `footer.html` override:

``` jinja2
{% if page.ancestors %}
  {{ page.ancestors[0] }}
{% endif %}
```

Feel free to share your final solution, when you managed to work something out, so others can benefit as well ☺️

**Reply by LizThompson93** [reply]
Solution if anyone needs it: Overwrite footer.html and modify the following lines:

```
                {% if page.previous_page.ancestors %}{{ page.previous_page.ancestors[0].title ~ ' - ' }}{% endif %}{{ page.previous_page.title }}


                {% if page.next_page.ancestors %}{{ page.next_page.ancestors[0].title ~ ' - ' }}{% endif %}{{ page.next_page.title }}
```

Thanks Martin, would never have known about page.ancestors!

Edited my code to reflect page.next_page and page.previous_page instead of simply page.
